### ISSUE TITLE
[2.12.5] invalid return type of `\Google\Service\Exception::getErrors()`

### ISSUE THREAD

**Issue by jacekkarczmarczyk** [googleapis/google-api-php-client#2270]
#### Environment details

  - OS: Win10
  - PHP version: 8.1
  - Package name and version: `google/apiclient@2.12.5`

#### Code example

```php
/** @var \Google\Service\Exception $exception */
array_map(fn (array $errorData) => $errorData['message'], $exception->getErrors());
```

Processing this code with phpstan results in the following error:
```
Parameter #1 $callback of function array_map expects (callable(string): mixed)|null, Closure(array): mixed given.
```

#### Details

According to phpdoc `Exception::getErrors` returns `array<string, string>`:
https://github.com/googleapis/google-api-php-client/blob/main/src/Service/Exception.php#L65

Result of this method is passed to the  `\Google\Task\Runner::allowedRetries()` method:
https://github.com/googleapis/google-api-php-client/blob/main/src/Task/Runner.php#L191

And the in `allowedRetries` it's used like this:
```php
$errors[0]['reason']
```
https://github.com/googleapis/google-api-php-client/blob/94e1964a7e0fd848eafdb1b485e0b68b9b5f08f1/src/Task/Runner.php#L281

Obviously that couldn't be possible if the `getErrors` method returned an array of strings

**Comment by ftipalek** [comment]
@jacekkarczmarczyk Thanks for reporting this 👍 
I got into same troubles yesterday evening when I was updating dependencies but did not make it open issue here yet.

**Comment by bshaffer** [comment]
Fixed in #2271
